SQLPractice/src/main/java/com/sql/topic5/ScriptRunner.java

```diff
@@ -0,0 +1,248 @@
+ 
+        private String delimiter = DEFAULT_DELIMITER;
+        private boolean fullLineDelimiter = false;
+ 
+        /**
+         * Default constructor
+         */
+        public ScriptRunner(Connection connection, boolean autoCommit,
+                        boolean stopOnError) {
+                this.connection = connection;
+                this.autoCommit = autoCommit;
+                this.stopOnError = stopOnError;
+        }
+ 
+        public void setDelimiter(String delimiter, boolean fullLineDelimiter) {
+                this.delimiter = delimiter;
+                this.fullLineDelimiter = fullLineDelimiter;
+        }
+ 
+        /**
+         * Setter for logWriter property
+         *
+         * @param logWriter
+         *            - the new value of the logWriter property
+         */
+        public void setLogWriter(PrintWriter logWriter) {
+                this.logWriter = logWriter;
+        }
+ 
+        /**
+         * Setter for errorLogWriter property
+         *
+         * @param errorLogWriter
+         *            - the new value of the errorLogWriter property
+         */
+        public void setErrorLogWriter(PrintWriter errorLogWriter) {
+                this.errorLogWriter = errorLogWriter;
+        }
+ 
+        /**
+         * Runs an SQL script (read in using the Reader parameter)
+         *
+         * @param reader
+         *            - the source of the script
+         */
+        public void runScript(Reader reader) throws IOException, SQLException {
+                try {
+                        boolean originalAutoCommit = connection.getAutoCommit();
+                        try {
+                                if (originalAutoCommit != this.autoCommit) {
+                                        connection.setAutoCommit(this.autoCommit);
+                                }
+                                runScript(connection, reader);
+                        } finally {
+                                connection.setAutoCommit(originalAutoCommit);
+                        }
+                } catch (IOException e) {
+                        throw e;
+                } catch (SQLException e) {
+                        throw e;
+                } catch (Exception e) {
+                        throw new RuntimeException("Error running script.  Cause: " + e, e);
+                }
+        }
+ 
+        /**
+         * Runs an SQL script (read in using the Reader parameter) using the
+         * connection passed in
+         *
+         * @param conn
+         *            - the connection to use for the script
+         * @param reader
+         *            - the source of the script
+         * @throws SQLException
+         *             if any SQL errors occur
+         * @throws IOException
+         *             if there is an error reading from the Reader
+         */
+        private void runScript(Connection conn, Reader reader) throws IOException,
```

Comment:
Medium: this method is huge and is difficult to understand or follow. Next time divide the method in several methods with full described names.